### PR TITLE
Server class docs

### PR DESCRIPTION
**Pull request by krlv**
Missing piece for #226

### CODE REVIEW

File: docs/classes/server.md [hunk 1]
```diff
@@ -3,71 +3,103 @@ title: Server
...
+
+Starts the server.
+
+## `stop(int $timeout = 3000ms): Promise`
+
+Stops the server. `$timeout` is the number of milliseconds to allow clients to gracefully shutdown before forcefully closing.
+
+The returned `Promise` will resolve when the server has successfully been stopped.
+
+## `__debugInfo(): array`
```
Comment (kelunik) [inline docs/classes/server.md]: IMO `__debugInfo()` shouldn't be documented or explicitly documented as not being public API.

**Comment by krlv** [comment]
@kelunik Moved info about debug output to the main description. Wdyt, should I remove it completely?

File: docs/classes/server.md [hunk 2]
```diff
@@ -3,71 +3,99 @@ title: Server
...
-{:.warning}
-> This page needs an update.
-
 * Table of Contents
 {:toc}
 
 The `Server` instance controls the whole listening and dispatches the parsed requests.
 
-## `attach(ServerObserver)`
+The `Server` class has [custom debug output](http://php.net/manual/en/language.oop5.magic.php#object.debuginfo):
```
Comment (kelunik) [inline docs/classes/server.md]: I wouldn't remove it completely, but remove the output and add: custom debug output, which isn't part of the public API and which shouldn't be relied upon.

Comment (krlv) [reply docs/classes/server.md]: Fixed.

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**